clinica/utils/freesurfer.py

```diff
@@ -218,289 +284,169 @@ def extract_image_id_from_longitudinal_segmentation(freesurfer_id):
-    value_list = list(df["Volume_mm3"]) + list(secondary_stats_dict.values())
-    write_tsv_file(
-        os.path.join(output_dir, prefix + "_parcellation-wm_volume.tsv"),
-        key_list,
-        "volume",
-        value_list,
-    )
+    Parameters
+    ----------
+    line : str
+        The line of the stat file to analyze.
 
+    info : InfoType
+        The type of information we are looking for.
 
-def write_tsv_file(out_filename, name_list, scalar_name, scalar_list):
-    """Write a .tsv file with list of keys and values.
+    Returns
+    -------
+    region : str
+        The region name for this line.
 
-    Args:
-        out_filename (string): name of the .tsv file
-        name_list (list of string): list of keys
-        scalar_name (string): 'volume', 'thickness', 'area' or
-            'meanCurv'. Not used for now. Might be used as part of a
-            pandas data frame
-        scalar_list (list of float): list of values corresponding to the keys
+        .. note::
+            This will be an empty string if the line does not contain
+            specific keywords related to the type of information.
+
+    value : str
+        The corresponding statistics value.
+
+        .. note::
+            This will be an empty string if the line does not contain
+            specific keywords related to the type of information.
     """
-    import warnings
+    info_to_keywords = {
+        InfoType.VOLUME: ["volume", "Volume"],
+        InfoType.AREA: ["area", "Area"],
+        InfoType.THICKNESS: ["thickness", "Thickness"],
+    }
+    word_list = line.replace(",", "").split()
+    if any(x in word_list for x in info_to_keywords[info]):
+        return word_list[2], word_list[-2]
+    return "", ""
+
+
+def get_secondary_stats(stats_filename: PurePath, info_type: InfoType) -> pd.DataFrame:
+    """Read the 'secondary' statistical info from .stats file.
+
+    Extract the information from .stats file that is commented out
+    (lines starting with '# Measure' prefix) and does not appear in the
+    table at the end of the document.
+
+    Parameters
+    ----------
+    stats_filename : PurePath
+        Path to the .stats file.
 
-    import pandas
+    info_type : InfoType
+        Type of information to read from .stats file.
 
-    try:
-        data = pandas.DataFrame({"label_name": name_list, "label_value": scalar_list})
-        data.to_csv(out_filename, sep="\t", index=False, encoding="utf-8")
-    except Exception as exception:
-        warnings.warn("Impossible to save {0} file".format(out_filename))
-        raise exception
+    Returns
+    -------
+    secondary_stats_dict : pd.DataFrame
+        DataFrame containing the regions of the brain (in a column named 'label_name'),
+        and the corresponding volume/thickness/area values depending on the
+        input info type (in a column named 'label_value').
+    """
+    if info_type == InfoType.MEANCURV:
+        return pd.DataFrame()
```

Comment:
The docstring says the function returns a dataframe with a certain shape (`label_name` and `label_value` columns), yet it can also return an empty dataframe not respecting this shape.

I wonder whether we would not be better off returning `None` and make the signature `-> Optional[pd.DataFrame]`. Unless there is an implementation detail that justifies it of course.

Reply:
I followed your suggestion to return `None`. To be honest, I'm not really sure to understand why mean curvature is considered as a special case since nothing is ever done for this type of information. I kind of feel why we should totally get rid of it... 🤔

Reply:
I agree. I guess we'll have it figured out when converting it to Pydra.